DB/src/main/java/io/deephaven/db/v2/utils/RuntimeMemory.java

```diff
@@ -88,91 +107,84 @@ public static RuntimeMemory getInstance() {
             freeMemory = totalMemory = totalCollections = totalCollectionTimeMs = 0L;
         }
 
+        /**
+         * Copy all the sample data in this instance to the corresponding fields in the provided instance.
+         *
+         * @param s destination for the copied data
+         */
         void copy(final Sample s) {
             freeMemory = s.freeMemory;
             totalMemory = s.totalMemory;
             totalCollections = s.totalCollections;
             totalCollectionTimeMs = s.totalCollectionTimeMs;
         }
-
-    }
-
-    public void read(Sample buf) {
-        maybeUpdateValues();
-        buf.freeMemory = lastFreeMemory;
-        buf.totalMemory = lastTotalMemory;
-        buf.totalCollections = totalCollections;
-        buf.totalCollectionTimeMs = totalCollectionTimeMs;
-    }
-
-    /**
-     * See {@link Runtime#maxMemory()}.
-     */
-    long getMaxMemory() {
-        return maxMemory;
     }
 
     /**
-     * If we are past the deadline for fetching free memory values, fetch the values.
+     * Read last collected samples.
      *
-     * Additionally, if we are past the deadline for logging; log the values.
+     * @param buf a user provided buffer object to store the samples.
      */
-    private void maybeUpdateValues() {
+    public void read(final Sample buf) {
         final long now = System.currentTimeMillis();
-        if (now >= nextCheck) {
-            lastFreeMemory = runtime.freeMemory();
-            lastTotalMemory = runtime.totalMemory();
-            nextCheck = now + cacheInterval;
+        Snapshot snapshot = currSnapshot;
+        if (now >= snapshot.nextCheck) {
+            snapshot = nextSnapshot;
+            snapshot.lastFreeMemory = runtime.freeMemory();
+            snapshot.lastTotalMemory = runtime.totalMemory();
+            snapshot.nextCheck = now + cacheInterval;
             long collections = 0;
             long collectionsMs = 0;
             for (final GarbageCollectorMXBean gcBean : gcBeans) {
                 collections += gcBean.getCollectionCount();
                 collectionsMs += gcBean.getCollectionTime();
             }
-            totalCollections = collections;
-            totalCollectionTimeMs = collectionsMs;
+            snapshot.totalCollections = collections;
+            snapshot.totalCollectionTimeMs = collectionsMs;
+            nextSnapshot = currSnapshot;
+            currSnapshot = snapshot;
         }
-        if (logInterval > 0 && now >= nextLog) {
-            log.info().append("Jvm Heap: ").append(commaFormat.format(lastFreeMemory)).append(" Free / ")
-                    .append(commaFormat.format(lastTotalMemory)).append(" Total (")
+        buf.freeMemory = snapshot.lastFreeMemory;
+        buf.totalMemory = snapshot.lastTotalMemory;
+        buf.totalCollections = snapshot.totalCollections;
+        buf.totalCollectionTimeMs = snapshot.totalCollectionTimeMs;
+        if (logInterval > 0 && now >= snapshot.nextLog) {
```

Comment:
I wonder if we should just delete this spammy crap now that we have a log table. It shows up at unpredictable intervals, anyway.